server/src/main/java/io/unitycatalog/server/UnityCatalogServer.java

```diff
@@ -258,6 +241,7 @@ public static void main(String[] args) {
 
   public void start() {
     LOGGER.info("Starting server...");
+    MetastoreRepository.getInstance().initMetastoreIfNeeded();
```

Comment:
Should we cache the response of `initMetastoreIfNeeded` since there's no way it can be modified once set?